File: 452.用最少数量的箭引爆气球/Solution.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        int ans = 1;
        Arrays.sort(points, (a,b) -> Integer.compare(a[0], b[0]));
        for(int i = 1; i < points.length; i++){
            if(points[i][0] > points[i-1][1]) ans++;
            else{
                points[i][1] = Math.min(points[i-1][1], points[i][1]);
            }
        }
        return ans;
    }
}

File: 452.用最少数量的箭引爆气球/Solution.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        # 排序后从前往后找重叠的气球，后面的气球的左端小于等于第一个气球的右端说明重叠，计算重叠数即使答案
        ans = 1 # 至少需要一支箭
        points.sort(key=lambda x: x[0])
        for i in range(1, len(points)):
            if points[i][0] > points[i-1][1]: # 说明不重叠，前面的部分需要一支箭
                ans += 1
            else:
                points[i][1] = min(points[i-1][1], points[i][1]) # 更新重叠气球最小右边界
        return ans